### PR TITLE
docs(cli): add info about development on github codespaces

### PR DESCRIPTION
**Pull request by vedantmgoyal2009**
### Motivation

* https://github.com/facebook/docusaurus/issues/6865#issuecomment-1062930085

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

### Related PRs

* https://github.com/facebook/docusaurus/issues/6865

### CODE REVIEW

**Comment by vedantmgoyal2009** [comment]
@facebook-github-bot recheck now.

**Comment by Josh-Cena** [comment]
@vedantmgoyal2009 The CLA bot updates the PR status automatically. It doesn't get triggered by anything.

**Comment by vedantmgoyal2009** [comment]
I do not have any problem in moving it to CLI docs but am just putting my thoughts here.

I think there are a considerable number of developers on GitHub that uses and even provide support for development on Codespaces and `.devcontainer` files for more easy development. For instance, I can remember of Oh-my-posh (https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md#codespaces--devcontainer-development-environment)

**Comment by Josh-Cena** [comment]
Yes, but that's `CONTRIBUTING`, which is developer-facing, not user-facing.

We have a lot of casual users who don't use these toolchains—they have a text editor, Node and NPM, and that's it. We don't want to clutter the installation guide (and any introductory content) with these niche use-cases. So yes, moving to CLI docs is the most sensible.

**Comment by vedantmgoyal2009** [comment]
I guess a section under "Advanced Guides" would make more sense?

**Comment by Josh-Cena** [comment]
No, because "advanced" is relative to users' understanding of _our_ infrastructure, not relative to their toolchain's advancedness :)

File: website/docs/cli.md [hunk 1]
```diff
@@ -50,6 +50,16 @@ Please note that some functionality (for example, anchor links) will not work in
 
 :::
 
+:::info Running on GitHub Codespaces
```
Comment (slorber) [inline website/docs/cli.md:53]: ok to add this here, but I'd like the block to be quite generic

It does not apply only to codespace, and this option is also available in other frameworks: CRA, Gatsby, NextJS...

See for example Next.js doc: https://nextjs.org/docs/api-reference/cli#development

**Comment by vedantmgoyal2009** [comment]
Thank you🎉🎉

**Pull request merged**